crates/fluvio-stream-model/src/core.rs

```diff
@@ -107,6 +117,12 @@ mod context {
         pub fn set_owner(&mut self, ctx: C) {
             self.owner = Some(ctx);
         }
+        pub fn children(&self) -> &HashMap<String, Vec<C>> {
+            &self.children
+        }
+        pub fn set_children(&mut self, children: HashMap<String, Vec<C>>) {
```

Comment:
Since this is specific to `local` metadata, may be better to put it in local storage rather than put this into common.  This feels like it is leaking abstraction since parent-child relationship should be tracked by storage

Reply:
It would be weird that Child -> Parent abstraction resides in `MetadataStoreObject` and the other way `Parent->Child` is in another place. It would not help for clarity and readability of the code.

Reply:
Currently, this call is only used by LocalStorage.  This leads to leak abstraction and inconsistency implementations.   Suggestions to work around this:
- Provide way to put "custom" object in the context so you can whatever you need do.  So something like: 
```
pub struct MetadataContext<C> {
        item: C,
        owner: Option<C>,
        custom: C::CustomContext
    }
```
- Or make metadata dynamic trait do downcast as necessary

-  Move all children relationship to Metadata and get rid of context
```
pub trait MetadataItem {
     ...
     owner()...
     setOwner(),
     getChildren()
     setChildren()
}
```

That way, you could implement

Reply:
I chose the third option: moved all to `MetadataItem` and removed them from `MetadataContext`, but I kept `MetadataContext` itself so far to reduce the number of changes in one PR and the context might be beneficial in the future.

Reply:
Great